### DISCUSSION TITLE
Use some of the CLI arguments inside my python code

### DISCUSSION THREAD

**Discussion by hanserasmus** [openai/whisper#1951] [Q&A] [answered]
Hi all

I am sorry if this is a stupid question. I cannot seem to find any documentation on how to do this, or understand from the source code how to implement what I want to achieve. I want to, amongst other things, limit the number of threads whisper is allowed to use, as it will be running on  a server that shares resources with other applications. So there are a couple of things that is available on the CLI, but does not seem to be available to the `transcribe` function. If you take the example below:

```
model = whisper.load_model("large")
result = model.transcribe("audio.mp3")
print(result["text"])
```
I would like to change it so that:
- it limits the whisper application to  2 threads
- it needs to make use of a model directory where the models have been downloaded into already
- it needs to return not text, but rather returning `srt` and `vtt`. 

Any possible examples I can look at to achieve what I want to achieve? TIA for anyone willing to help.

**Comment by ryanheise** [comment, marked as answer]
You can pass those arguments like this:

```python
import whisper
from whisper.utils import get_writer

audio_path = "audio.mp3"
model_dir = "path/to/dir"
output_dir = "path/to/dir"

torch.set_num_threads(2)
model = whisper.load_module("large", download_root=model_dir)
result = model.transcribe(audio_path, word_timestamps=True)
srt_writer = get_writer("srt", output_dir)
vtt_writer = get_writer("vtt", output_dir)
srt_writer(result, audio_path)
vtt_writer(result, audio_path)
```

**Reply by hanserasmus** [reply]
Thank you so much @ryanheise . I will try this immediately.

**Reply by hanserasmus** [reply]
@ryanheise I get an error:
```
Traceback (most recent call last):
  File "/home/hans/SSD/Python_Projects/nextcloud-transcription/transcribe.py", line 42, in <module>
    model = whisper.load("large", download_root=model_dir)
AttributeError: module 'whisper' has no attribute 'load'
```
My full python script is as follows:

```
import requests
import rq
import redis
import whisper
import torch
from parse_config import parse_config_redis
from whisper.utils import get_writer

def calculate_threads():
    try:
        res = open('/proc/cpuinfo').read().count('processor\t:')
        if res > 0:
            count = int(res / 2)
            return count
    except IOError:
        pass

r_host, r_port, r_db = parse_config_redis()
redis_conn = redis.Redis(host=r_host, port=r_port, db=r_db, decode_responses=True)

def is_redis_available(redis_conn):
    try:
        redis_conn.ping()
        print("Successfully connected to redis")
    except (redis.exceptions.ConnectionError, ConnectionRefusedError):
        print("Redis connection error!")
        return False
    return True

if is_redis_available(redis_conn):
    print("Yay!")

thread_count = calculate_threads()

audio_path = "File1.ogg"
model_dir = "/tmp/transcribe/models"
output_dir = "/tmp/transcribe/results"

torch.set_num_threads(thread_count)
model = whisper.load("large", download_root=model_dir)
result = model.transcribe(audio_path, word_timestamps=True)
srt_writer = get_writer("srt", output_dir)
vtt_writer = get_writer("vtt", output_dir)
srt_writer(result, audio_path)
vtt_writer(result, audio_path)
```
Sorry to pester you with this.

**Reply by ryanheise** [reply]
Sorry about that, I meant `load_model`. I have updated my answer.

**Reply by hanserasmus** [reply]
Updated my code to now be:
`model = whisper.load_module(("large", download_root=model_dir))`

but I get:

![image](https://github.com/openai/whisper/assets/24249616/116a6e3a-0fb4-4773-8206-be4bbefc4160)

Sorry I know I am probably supposed to figure this out myself.

**Reply by ryanheise** [reply]
Can you try copying and pasting from my answer? Looks like you may have mistyped it.

**Reply by hanserasmus** [reply]
I am so sorry @ryanheise that was so stupid of me. But, looking at your answer again, I am not sure if the github page is not updating properly or what, because this is what I see on my side:

![image](https://github.com/openai/whisper/assets/24249616/05ff7504-f82e-43ea-9f26-7c6065b8f687)

But later on you said you meant `load_model`. 

So not sure what is going on but anyways, updated my code to now read `model = whisper.load_model(("large", download_root=model_dir))` and all that pycharm is left complaining about is:

![image](https://github.com/openai/whisper/assets/24249616/267599a7-600c-4f03-9978-99e0ad4a5620)

**Reply by hanserasmus** [reply]
Ah the single parentheses is the answer it seems. 
`model = whisper.load_model("large", download_root=model_dir)`

**Reply by ryanheise** [reply]
My`(` key must have been over excited :-)

**Reply by hanserasmus** [reply]
Will see how this goes, thank you SO MUCH for your kindness and patience.
![image](https://github.com/openai/whisper/assets/24249616/c1951ff5-2d80-45ad-bd87-b68a5a06e7c1)

**Comment by phineas-pta** [comment]
see how whisper pass CLI args: https://github.com/openai/whisper/blob/main/whisper/transcribe.py#L511
